store/sqlstore/channel_store_categories.go

```diff
@@ -565,33 +588,33 @@ func (s SqlChannelStore) UpdateSidebarCategoryOrder(userId, teamId string, categ
 			}
 		}
 		if !found {
-			return model.NewAppError("SqlPostStore.UpdateSidebarCategoryOrder", "store.sql_channel.sidebar_categories.app_error", nil, "Cannot update category order, passed list of categories contains unrecognized category IDs", http.StatusBadRequest)
+			return store.NewErrInvalidInput("SidebarCategories", "id", fmt.Sprintf("%v", categoryOrder))
 		}
 	}
 
-	if appErr := s.updateSidebarCategoryOrderT(transaction, userId, teamId, categoryOrder); appErr != nil {
-		return appErr
+	if err = s.updateSidebarCategoryOrderT(transaction, userId, teamId, categoryOrder); err != nil {
+		return err
 	}
 
 	if err = transaction.Commit(); err != nil {
-		return model.NewAppError("SqlChannelStore.UpdateSidebarCategoryOrder", "store.sql_channel.sidebar_categories.commit_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return errors.Wrap(err, "commit_transaction")
 	}
 
 	return nil
 }
 
-func (s SqlChannelStore) UpdateSidebarCategories(userId, teamId string, categories []*model.SidebarCategoryWithChannels) ([]*model.SidebarCategoryWithChannels, *model.AppError) {
+func (s SqlChannelStore) UpdateSidebarCategories(userId, teamId string, categories []*model.SidebarCategoryWithChannels) ([]*model.SidebarCategoryWithChannels, error) {
 	transaction, err := s.GetMaster().Begin()
 	if err != nil {
-		return nil, model.NewAppError("SqlChannelStore.UpdateSidebarCategory", "store.sql_channel.sidebar_categories.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, errors.Wrap(err, "begin_transaction")
 	}
 	defer finalizeTransaction(transaction)
 
 	updatedCategories := []*model.SidebarCategoryWithChannels{}
 	for _, category := range categories {
-		originalCategory, appErr := s.GetSidebarCategory(category.Id)
-		if appErr != nil {
-			return nil, model.NewAppError("SqlPostStore.UpdateSidebarCategory", "store.sql_channel.sidebar_categories.app_error", nil, appErr.Error(), http.StatusInternalServerError)
+		originalCategory, nErr := s.GetSidebarCategory(category.Id)
```

Comment:
If err is already of type `error`, is there a reason to do `nErr` again?

Reply:
It seems you don't like sentences like `var originalCategory *model.SidebarCategoryWithChannels` :hand_over_mouth: so I decided to create a new err to avoid that, but as you prefer.

Reply:
Oh, in that case, you can just do err2. Using nErr or appErr has special meaning.

Reply:
Got it!